### PR TITLE
Publish to multiple channels - Part 2

### PR DESCRIPTION
**Pull request by epananth**
Publish to multiple channels-> https://github.com/dotnet/arcade/issues/6669

Test Promotion Build -> https://dnceng.visualstudio.com/internal/_build/results?buildId=940988&view=logs&j=ba23343f-f710-5af9-782d-5bd26b102304&t=6e277ba4-1c1e-552d-b96f-db0aeb4be20a

Couldn't get arcade ci to be successful cos need to get this PR merged before -> https://github.com/dotnet/arcade-services/pull/1502

### CODE REVIEW

**Comment by epananth** [comment]
This also solves the retry failure for blobs, I reran the maestro promotion pipeline to add the same build to general testing channel, and it did not give the asset is already present error 

https://dnceng.visualstudio.com/internal/_build/results?buildId=941024&view=logs&j=ba23343f-f710-5af9-782d-5bd26b102304&t=6e277ba4-1c1e-552d-b96f-db0aeb4be20a

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/BlobFeedAction.cs [hunk 1]
```diff
@@ -190,8 +190,9 @@ public async Task PublishToFlatContainerAsync(IEnumerable<ITaskItem> taskItems,
                 }
                 else
                 {
+                    FileStream stream = new FileStream(item.ItemSpec, FileMode.Open, FileAccess.Read, FileShare.Read);
```
Comment (mmitche) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/BlobFeedAction.cs:193]: Should be 
```
using (FileStream stream ...)
{

}
```

**Comment by epananth** [comment]
Successful arcade build -> https://dnceng.visualstudio.com/internal/_build/results?buildId=943562&view=results

**Comment by epananth** [comment]
Going to merge after arcade-services roll out, cos currently the darc-verison is hard coded.

File: eng/common/post-build/publish-using-darc.ps1 [hunk 2]
```diff
@@ -16,8 +16,7 @@ param(
 
 try {
   . $PSScriptRoot\post-build-utils.ps1
-  # Hard coding darc version till the next arcade-services roll out, cos this version has required API changes for darc add-build-to-channel
-  $darc = Get-Darc "1.1.0-beta.20418.1"
+  . $PSScriptRoot\..\darc-init.ps1
```
Comment (riarenas) [inline eng/common/post-build/publish-using-darc.ps1:19]: We should still use the Get-Darc function. If you don't pass a version it calls darc-init without parameters.

**Comment by epananth** [comment]
Test with latest darc  -> https://dnceng.visualstudio.com/internal/_build/results?buildId=951751&view=logs&j=226748d0-f812-5437-d3f0-2dd291f5666e&t=bad11196-972e-5d03-45a8-9db526506073

**Review by mmitche — APPROVED** [review]

**Pull request merged**